swipe-button/src/main/java/com/ebanx/swipebtn/SwipeButton.java

```diff
@@ -138,9 +141,55 @@ public void setHasActivationState(boolean hasActivationState) {
         this.hasActivationState = hasActivationState;
     }
 
+    /**
+     * Enable or Disable Swipe from code
+     */
+    public void enableSwipeAction(boolean enabled) {
+        setAlphaState(enabled);
+        if (!enabled) {
+            setOnTouchListener(new OnTouchListener() {
+                @Override
+                public boolean onTouch(View v, MotionEvent rawEvent) {
+                    return false;
+                }
+            });
+        } else {
+            setOnTouchListener(getButtonTouchListener());
+        }
+    }
+
+    /**
+     * Set alpha level
+     *
+     * @param enabled
+     */
+    private void setAlphaState(boolean enabled) {
+        if (!enabled) {
+            buttonBackground.setAlpha(100);
+//            trailBackground.setAlpha(100);
+            layoutBackground.setAlpha(100);
+        } else {
+            buttonBackground.setAlpha(255);
+//            trailBackground.setAlpha(255);
+            layoutBackground.setAlpha(255);
+        }
+    }
+
+    public void setButtonColor(int color) {
+        buttonBackground.setColorFilter(ContextCompat.getColor(mContext, color), PorterDuff.Mode.SRC_ATOP);
+    }
+
+    public void setTrailBgColor(int color) {
+        trailBackground.setColorFilter(ContextCompat.getColor(mContext, color), PorterDuff.Mode.SRC_ATOP);
+    }
+
+    public void setLayoutBgColor(int color) {
+        layoutBackground.setColorFilter(ContextCompat.getColor(mContext, color), PorterDuff.Mode.SRC_ATOP);
+    }
+
     private void init(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
         hasActivationState = true;
-
+        mContext = context;
```

Comment:
There no need to have a property for instance. You can access the Context using getContext()